core/course_controller/CourseController.py

```diff
@@ -56,15 +83,25 @@ def save_course(course_data: CourseFormData, course_id: int | None = None) -> st
         return str(course.id)
 
     @staticmethod
-    def get_course(course_id: int) -> CourseOverview:
+    def get_course(course_id: int, semester_name: str) -> CourseOverview:
         """
         Pre-conditions: course_id provided is a valid course id
         Post-conditions: Returns an object containing course, sections, and assignment
             information for object in the Courses table with the given course_code.
         Side-effects: N/A
         """
+        #Get semester instance using semester_name
+        #Refactored
+        try:
+            semester = Semester.objects.get(semester_name= semester_name)
```

Comment:
why get the semester here. it would throw an error if we search using "semester__semester_name" when getting the course anyways.